realm/realm-library/src/androidTest/java/io/realm/RealmQueryTests.java

```diff
@@ -769,6 +774,190 @@ public void isNull_notNullableFields() {
         }
     }
 
+    // Querying nullable PrimaryKey
+    @Test
+    public void equalTo_nullPrimaryKeys() {
+        final long SECONDARY_FIELD_NUMBER = 49992417L;
```

Comment:
Lots of duplicated tests in `RealmJsonTests` are now refactored into only three parameterized tests in [`RealmJsonNullPrimaryKeyTests`](https://github.com/realm/realm-java/blob/stk/null-primarykey/realm/realm-library/src/androidTest/java/io/realm/RealmJsonNullPrimaryKeyTests.java).

For those tests in `RealmQueryTests`, I'd plead for your reconsideration on leaving them as is. I agree that they look mundane, and so slightly alternate from one to another. Nonetheless, I have to point that the seemingly replicated parts are from the similarities of queries, not the purpose of tests themselves. 

I _believe_ each one of them has to be throughly checked.